src/simplewallet/simplewallet.cpp

```diff
@@ -2360,6 +2360,25 @@ bool simple_wallet::start_mining_for_rpc(const std::vector<std::string> &args)
   if (!try_connect_to_daemon())
     return true;
 
+  bool ok = true;
+  if(args.size() >= 1)
+  {
+    uint16_t num = 1;
+    ok = string_tools::get_xtype_from_string(num, args[0]);
+    ok = ok && 1 <= num;
+    m_rpc_payment_threads = num;
+  }
+  else
+  {
+    m_rpc_payment_threads = 1;
+  }
+
+  if (!ok)
+  {
+    PRINT_USAGE(USAGE_START_MINING_FOR_RPC);
+    return true;
+  }
```

Comment:
This looks like if you run:
> start_mining_for_rpc 1
> start_mining_for_rpc 0

Then second one will fail as ok will be false, but m_rpc_payment_threads will still be set and will find its way into the call to search_for_rpc_payment later. Admittedly it's a "don't do that then" situation :)

Reply:
Well spotted.